Library integration backend

danielcharry opened the pull request:
## Summary ##

**Trello card**: https://trello.com/c/vTPbpTfS/11-vista-de-biblioteca

Conexión del proyecto con el API de WBooks en la pantalla de Library.

danielcharry committed:
Vista de libreria conectada al api

danielcharry committed:
vista de detalle libro

danielcharry committed:
Titulo del navbar en bookDetail

DiegoQuiros94 commented on DcProject/BookDetailController.swift:
```diff
@@ -0,0 +1,55 @@
+private extension BookDetailController {
+    
+    private func bindViewModel() {
+        
```
Supongo que esta funcion la dejaste vacia porque la vas a necesitar en un futuro, en ese caso estaria bueno que dejes un comentario aclarandolo. si directamente no la necesitas podes borrarla.

DiegoQuiros94 commented on DcProject/BookRepository.swift:
```diff
@@ -0,0 +1,24 @@
+    
+    public func getBooks() -> SignalProducer<[Book], RepositoryError> {
+        
+        return performRequest(method: .get, path: "/books") { response in
```
Como buena practica las cosas "hardcodeadas" las ponemos en una constante global dentro de la clase, de esa forma si hay que cambiar algo son mas fáciles de encontrar. en este caso podrías hacer eso con el string "/books". Ademas como es un repositorio lo mas probable es que tengas varios path mas, por lo que podes agruparlos todos en una extension. algo así:

```
// MARK: - Paths
private extension BookRepository {
    static let BooksPath: String = "/books"
}
```

y el performRequest te quedaria:

`performRequest(method: .get, path: BookRepository.BooksPath)`

DiegoQuiros94 commented on DcProject/LibraryController.swift:
```diff
@@ -39,6 +38,9 @@ final class LibraryController: UIViewController {
         configTable()
         confingNavBar()
+        requestBooks()
+        
```
borra la linea extra

DiegoQuiros94 commented on DcProject/LibraryController.swift:
```diff
@@ -55,29 +57,41 @@ final class LibraryController: UIViewController {
+        _viewModel.getBooks().startWithResult { result in
+            switch result {
+            case .success(let books):
+                print("Request Success!")
```
los prints en general los usamos para debuguear cosas y despues los borramos, en los PR no deberia quedar ningun print. Si te parece util que algo se loguee siempre en la consola podes usar NSLog o algun framework para loguear eventos

DiegoQuiros94 commented on DcProject/TabBarController.swift:
```diff
@@ -33,6 +37,13 @@ final class TabBarController: UITabBarController, UITabBarControllerDelegate, UI
+        let bookDetailController = BookDetailController(viewModel: BookDetailViewModel())
+        let navBookDetail = UINavigationController(rootViewController: bookDetailController)
+        bookDetailController.tabBarItem = UITabBarItem(
+            title: "BOOKDETAIL",
```
este titulo deberia estar localizado

DiegoQuiros94 commented on DcProject/TabBarController.swift:
```diff
@@ -33,6 +37,13 @@ final class TabBarController: UITabBarController, UITabBarControllerDelegate, UI
+        let navBookDetail = UINavigationController(rootViewController: bookDetailController)
+        bookDetailController.tabBarItem = UITabBarItem(
+            title: "BOOKDETAIL",
+            image: UIImage(named: "ICON-WISHLIST".localized()),
```
estos strings no tienen que estar localizados (a menos que sea un caso particular en que la imagen a utilizar cambie segun el idioma del telefono, pero no es el caso).

danielcharry committed:
Update BookDetailController.swift

Se elimina la funcion bindViewModel() ya que no esta en uso

danielcharry committed:
Update BookRepository.swift

Se crea el path para traer los libros del api

danielcharry committed:
Update LibraryController.swift

Se remplazan los print por NSLog

danielcharry committed:
Update Localizable.strings

Se elimina localizables de los iconos

danielcharry committed:
Update TabBarController.swift

Se elimina el bookDetail del tabBar y se cambian los localizables de los iconos por strings
